File: src/main/java/ro/ubb/dp1819/micu/emerson/domain/coffee/Cappuccino.java
package ro.ubb.dp1819.micu.emerson.domain.coffee;

import ro.ubb.dp1819.micu.emerson.domain.Ingredient;

import java.util.Arrays;

public class Cappuccino extends Coffee{

    //here we add the basic ingredients for cappuccino
    public Cappuccino(){
        ingredients = Arrays.asList(new Ingredient(2.0f, "cups", "water"),
                new Ingredient(0.25f, "cups", "coffee-beans", "roasted"),
                new Ingredient(100.0f, "ml", "milk", "foamed")
        );
    }

    @Override
    public String toString() {
        return "Cappuccino{"+ingredients.toString()+"}";
    }
}

File: LabsSimonaBurlacu232/Lab1/Lab1/Lab1/Domain/Cappuccino.cs
using System.Collections.Generic;

namespace Lab1.Domain
{
    public class Cappuccino:ICoffee
    {
        public IList<Ingredient> ingredients;
        public static Recipe recipe;
        public static Recipe Recipe
        {
            get
            {
                if (recipe == null)
                {
                    recipe = new Recipe()
                    {
                        Ingredients = new List<Ingredient>()
                        {
                            new Ingredient(0.25, Unit.cups, IngredientType.coffeebeans, CoffeeFeature.roasted),
                            new Ingredient(2, Unit.cups, IngredientType.water, null),
                            new Ingredient(100, Unit.ml, IngredientType.milk, CoffeeFeature.foamed),
                        }
                    };
                }

                return recipe;
            }
        }

        public Cappuccino()
        {
            ingredients = new List<Ingredient>();
        }

        public string GetCoffeeType()
        {
            return "Cappuccino";
        }
    }
}
